docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,459 @@
+Inactive Cells at the Top of the Water Column
+=============================================
+
+date: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+
+
+Summary
+-------
+
+For increased flexibility in representing ice-shelf cavities in MPAS-Ocean,
+we require a mechanism for inactivating cells in the water column that are
+above the ice draft (the depth of the ice shelf-ocean interface).  This
+capability would be similar to the disabled cells below the bathymetry.
+Currently, ice-shelf cavities use a vertical coordinate where top layer follows
+the ice draft, but pressure-gradient errors caused by steep, tilted layers
+place many restrictions on this approach (e.g. layers are thick and bathymetry
+is not well represented).  As a step toward a more flexible vertical coordinate
+that can better adapt to changing ice topography in time, we require a
+capability to disable top cells and thus mitigate the constraints imposed by
+the current terrain-following top coordinate.  A successful implementation will
+be able to run an MPAS-Ocean test case with ice-shelf cavities using a
+prescribed vertical coordinate with inactive cells at the top of the water
+column.  The test case will compute "surface" fluxes at the top active layer
+and distribute them from the top down into active layers.  All test cases where
+no top layers are inactive must produce results that remain bit-for-bit
+identical to the current implementation.
+
+
+Requirements
+------------
+
+Requirement: Inactive top cells
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+We require a mechanism for indicating that top cells in the water column are
+inactive and preventing computation on these cells.
+
+Requirement: Surface fields and fluxes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+MPAS-Ocean includes several fields that represent the surface values of 3D
+fields.  These fields must use the top active cell in a column rather than the
+first (potentially inactive) cell.  Similarly, the distribution of surface
+fluxes into the water column must begin from the top with the first active
+cell, not the first (potentially inactive) cell.
+
```

Comment:
Probably also need a requirement that boundary conditions are applied correctly as well. Thinking here about higher-order vertical interpolants in routines like tracer advection. Not only are loop limits changed, but some new edge cases appear if max-min is not large enough.

Reply:
That's a very good point, thanks!

Reply:
This is going to be hard to fully address in this particular part of the design.  Some of the work on preventing pathological topography (e.g. 2 adjacent cells where each has the minimum allowed number of layers, say 3, but the layers don't overlap at all or don't overlap enough in the vertical to allow any transport across that edge) would have to be on the side of creating and maintaining the vertical coordinate itself, which is beyond the scope of this design.